### PR TITLE
Add release diffing script

### PR DESCRIPTION
**Pull request by devongovett**
This adds a script that can be used to diff the packages to determine what changed since the last release. It also updates the logic in the versioning script to match.

We must diff each package individually, based on the git tag of the last release for that package. We previously just looked at the last tag on the whole repo, but this doesn't work if some packages are skipped during the last release.

### CODE REVIEW

**Comment by snowystinger** [comment]
o no, i never opened the PR for this work, womps https://github.com/adobe/react-spectrum/tree/check-dists

File: scripts/diff.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const exec = require('child_process').execSync;
+const spawn = require('child_process').spawnSync;
+const fs = require('fs');
+
+let packages = JSON.parse(exec('yarn workspaces info --json').toString().split('\n').slice(1, -2).join('\n'));
+let uuid = require('uuid')();
+
+let ksdiff = process.argv.includes('--ksdiff');
```
Comment (snowystinger) [inline scripts/diff.js:8]: use yargs? we already have it in the repo

Comment (devongovett) [reply scripts/diff.js:8]: seems like overkill for two flags, no? 😉

Comment (snowystinger) [reply scripts/diff.js:8]: yeah, but it includes the nice help flag I think listing out the options

Comment (dannify) [reply scripts/diff.js:8]: Follow up if we add more flags.

**Review by dannify — APPROVED** [review]

**Comment by dannify** [comment]
Rob verbally approved

**Pull request merged**